roles/provision-keycloak-apb/tasks/configure-keycloak.yml

```diff
@@ -47,47 +39,19 @@
     status_code: 201
   when: keycloak_realm_exists.status == 404
 
-- name: Create {{ generated_username.stdout }} client in realm {{ KEYCLOAK_REALM }}
-  uri:
-    url: "{{ KEYCLOAK_HOST }}/auth/admin/realms/{{ KEYCLOAK_REALM }}/clients"
-    method: POST
-    body: '{"id": "{{ generated_username.stdout }}", "secret": "{{ generated_password.stdout }}", "redirectUris":["http://localhost:*"], "webOrigins":["http://localhost:8100"], "publicClient": true }'
-    validate_certs: no
-    body_format: json
-    headers:
-      Authorization: "bearer {{ keycloak_auth_response.json.access_token }}"
-    status_code: 201
-
-- name: Get installation details
-  uri:
-    url: "{{ KEYCLOAK_HOST }}/auth/admin/realms/{{ KEYCLOAK_REALM }}/clients/{{ generated_username.stdout }}/installation/providers/keycloak-oidc-keycloak-json"
-    method: GET
-    validate_certs: no
-    headers:
-      Authorization: "bearer {{ keycloak_auth_response.json.access_token }}"
-    status_code: 200
-    return_content: yes
-  register: installation_text
-
-- set_fact: installation="{{ installation_text.content | from_json }}"
-
-- name: Create client config from template
-  template:
-    src: installation.json.j2
-    dest: /tmp/installation.json
-
-- name: Retrieve client config
-  shell: cat /tmp/installation.json
-  register: installation_text
-
-- set_fact: installation="{{ installation_text.stdout | from_json }}"
-
 - name: Create keycloak configmap template
   template:
     src: configmap.yml.j2
     dest: /tmp/configmap.yaml
+  when: _apb_service_instance_id is defined
+
+- name: Create keycloak configmap template for tests
```

Comment:
test logic and test templates should not be added to provision, bind tasks

Reply:
This is needed for the build to pass as `_apb_service_instance_id` is not available when running `apb test`

Reply:
`apb test` should run `test.yml` playbook, and all templates and data should be stored accordingly. test logic cant be mixed with production

Reply:
@psturc @pb82 do you guys have any ideas for a way around this?

Reply:
@vchepeli We use this switch (`when _apb_service_instance_id is defined is defined`) in other APBs too (metrics and UPS). It's not ideal but at the moment we don't have a better solution. 

@psturc do we know why this parameter is not available at test-time? Is this a bug in the asb that we should open an issue for?

Reply:
@dimitraz I don't know if it's better but an alternative would be to make it so that `_apb_service_instance_id` is always defined:

```
- set_fact:
     _apb_service_instance_id: dummy
     when: _apb_service_instance_id is not defined
```

(have not tested this)

Reply:
@pb82 the reason why it's not available is that the `apb test` doesn't create a serviceinstance (which is created when you provision APB via catalog or mobile-cli)

I haven't asked anyone from ASB group if they plan to enhance the `apb test` with adding the serviceinstance, but we could potentially workaround this by running the tests in CI with mobile-cli.

Reply:
@psturc sounds like a good feature and PR